src/data/num/to_bits.lean

```diff
@@ -0,0 +1,150 @@
+/-
+Copyright (c) 2022 Praneeth Kolichala. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Praneeth Kolichala
```

Comment:
A lot of this code is just moved and comes from #3976, so the copyright and authors should be preserved:
```suggestion
Copyright (c) 2020 Pim Spelier. All rights reserved.
Released under Apache 2.0 license as described in the file LICENSE.
Authors: Pim Spelier, Daan van Gent, Praneeth Kolichala
```